### DISCUSSION TITLE
Run script and interact with it from another script

### DISCUSSION THREAD

**Discussion by jschwarty** [johnlindquist/kit#260] [Q&A] [answered]
This is a great example of running an existing script from another script:
https://github.com/johnlindquist/kit/discussions/49

But I am trying to track down how to run a script and interact with it programatically. For example, I would like to be able to programmatically call the `app-launcher` script and tell it what app to launch. Is this possible for scripts that are awaiting user input?

**Comment by jschwarty** [comment, marked as answer]
So I just found the answer by looking at the `focus-twitter` which has the following code:
```
await run('chrome-tab', 'twitter.com')
```
Looks like the second (and maybe more) args to the `run()` function will be the args. And further investigation into the `chrome-tab` script...
```
let {focusTab} = await kit('chrome')
let url = await arg('Enter url:')
focusTab(url)
```
reveals that it is awaiting user input (same as the `app-launcher`)!

*I had tried the second arg but it didn't appear to work. Then I ran the `app-launcher` script and checked the name of the app I was trying to launch and I was not using an exact match of the app name string. I used the same app name string from this and it worked! In my case I wanted to launch the Clockify Desktop app. My working script code looks like the following:*
```
await run('app-launcher', 'Clockify Desktop')
```

**Reply by jschwarty** [reply]
PS - I freakin' love Script Kit. Thanks a ton for creating and sharing this!

**Reply by johnlindquist** [reply]
Thanks @jschwarty! Glad you got it figured out.

The intention is that `run` mimics how you would run the script from the terminal. (Script Kit was originally terminal-only, so that's influenced the apis around args)

I really appreciate the kind words. They really help keep the motivation up 😊
